fdbserver/storageserver.actor.cpp

```diff
@@ -911,97 +939,6 @@ struct BusiestWriteTagContext {
 	    busiestWriteTagEventHolder(makeReference<EventCacheHolder>(busiestWriteTagTrackingKey)), lastUpdateTime(-1) {}
 };
 
-// A SSPhysicalShard represents a physical shard, it contains a list of keyranges.
-class SSPhysicalShard {
-public:
-	SSPhysicalShard(const int64_t id) : id(id) {}
-
-	void addRange(Reference<ShardInfo> shard);
-
-	// Remove the shard if a shard to the same pointer (ShardInfo*) exists.
-	void removeRange(Reference<ShardInfo> shard);
-
-	// Clear all shards overlapping with `range`.
-	void removeRange(KeyRangeRef range);
-
-	bool supportCheckpoint() const;
-
-	bool hasRange(Reference<ShardInfo> shard) const;
-
-	int size() const { return ranges.size(); }
-	// Public function to iterate over the ranges
-	std::vector<Reference<ShardInfo>>::const_iterator begin() const { return ranges.begin(); }
-
-	std::vector<Reference<ShardInfo>>::const_iterator end() const { return ranges.end(); }
-
-private:
-	const int64_t id;
-	std::vector<Reference<ShardInfo>> ranges;
-};
-
-void SSPhysicalShard::addRange(Reference<ShardInfo> shard) {
-	TraceEvent(SevVerbose, "SSPhysicalShardAddShard")
-	    .detail("ShardID", format("%016llx", this->id))
-	    .detail("Assigned", !shard->notAssigned())
-	    .detail("Range", shard->keys);
-	ASSERT(!shard->notAssigned());
-
-	removeRange(shard->keys);
-
-	ranges.push_back(shard);
-}
-
-void SSPhysicalShard::removeRange(Reference<ShardInfo> shard) {
-	TraceEvent(SevVerbose, "SSPhysicalShardRemoveShard")
-	    .detail("ShardID", format("%016llx", this->id))
-	    .detail("Assigned", !shard->notAssigned())
-	    .detail("Range", shard->keys);
-
-	for (int i = 0; i < this->ranges.size(); ++i) {
-		const auto& r = this->ranges[i];
-		if (r.getPtr() == shard.getPtr()) {
-			this->ranges[i] = this->ranges.back();
-			this->ranges.pop_back();
-			return;
-		}
-	}
-}
-
-void SSPhysicalShard::removeRange(KeyRangeRef range) {
-	TraceEvent(SevVerbose, "SSPhysicalShardRemoveRange")
-	    .detail("ShardID", format("%016llx", this->id))
-	    .detail("Range", range);
-	for (int i = 0; i < this->ranges.size();) {
-		const auto& r = this->ranges[i];
-		if (r->keys.intersects(range)) {
-			this->ranges[i] = this->ranges.back();
-			this->ranges.pop_back();
-		} else {
-			++i;
-		}
-	}
-}
-
-bool SSPhysicalShard::supportCheckpoint() const {
-	for (const auto& r : this->ranges) {
-		ASSERT(r->desiredShardId == this->id);
-		if (r->shardId != this->id) {
-			return false;
-		}
-	}
-	return true;
-}
-
-bool SSPhysicalShard::hasRange(Reference<ShardInfo> shard) const {
-	for (int i = 0; i < this->ranges.size(); ++i) {
-		if (this->ranges[i].getPtr() == shard.getPtr()) {
-			return true;
-		}
-	}
-
-	return false;
-}
-
 struct TenantSSInfo {
```

Comment:
What is TenantSSInfo used for? Thanks

Reply:
I believe it's used for representing the tenant states in SS. A SS could have multiple tenants. The state is managed here.